src/engine/source/_builder/expression.hpp

```diff
@@ -0,0 +1,447 @@
+#ifndef _EXPRESSION_H
+#define _EXPRESSION_H
+
+#include <algorithm>
+#include <functional>
+#include <memory>
+#include <sstream>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+#include <fmt/format.h>
+
+// namespace builder::internals
+// {
+
+// The Formula hierarchy is a description of the graph expression to be built,
```

Comment:
By separating the description from the expression you are able to use filters, map, and subscription, (the way it's done inside rxcppFactory) right?